uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/service/UniProtEntryService.java

```diff
@@ -119,12 +120,14 @@ protected UniProtQueryProcessorConfig getQueryProcessorConfig() {
     public UniProtKBEntry findByUniqueId(String accession, String fields) {
         try {
             List<ReturnField> fieldList = OutputFieldsParser.parse(fields, returnFieldConfig);
-            SolrRequest solrRequest =
-                    SolrRequest.builder()
-                            .query(ACCESSION + ":" + accession.toUpperCase())
-                            .rows(NumberUtils.INTEGER_ONE)
-                            .build();
+            accession = accession.toUpperCase();
+            SolrRequest solrRequest = buildSolrRequest(accession);
             Optional<UniProtDocument> optionalDoc = repository.getEntry(solrRequest);
+            if (accession.endsWith(CANONICAL_ISOFORM) && optionalDoc.isEmpty()) {
```

Comment:
Correct me if i am wrong.
accession will never be null
canonical isoform is always ends with xxx-1 means xxx-2 or xxx-3 never be canonical isoform
we have index solr with canonical isoform

Reply:
accession will never be null: correct, accession is a required parameter
Yes, basically the uniprot entry P12345 and P12345-1 are the same thing, the user can request both ways, and we need to return P12345. (first request will fail, and we remove the -1 and query again)
When we have entries that has canonical + isoforms (P21802, P21802-1, P21802-2, ...) (we are indexing it as 2 different entries in solr and voldemort), in this case, the first request will return it.
Does it make sense?